workflow-service/src/main/java/com/redhat/parodos/workflow/execution/service/WorkFlowServiceImpl.java

```diff
@@ -101,7 +106,9 @@ public WorkReport execute(WorkFlowRequestDTO workFlowRequestDTO) {
 			return new DefaultWorkReport(WorkStatus.FAILED, new WorkContext(), new Throwable(validationFailedMsg));
 		}
 
-		WorkContext workContext = workFlowDelegate.initWorkFlowContext(workFlowRequestDTO);
+		WorkFlowDefinition workFlowDefinition = workFlowDefinitionRepository.findFirstByName(workflowName);
```

Comment:
At some point, I think that we need to get rid of workflowDefinitionRepository from here, Does not make sense that from another domain we keep using internal definitions instead of DTO

Reply:
agree, I will change to getByName

Reply:
I think we should build the work relations from internal and not rely on the payload from UI.